submit-test-status/action.yml

```diff
@@ -33,58 +37,112 @@ runs:
       echo "-----"
       echo "Job name override: ${{ inputs.job_name_override }} (default job name: $GITHUB_JOB)"
       echo "BQ table name (for testing): ${{ inputs.big_query_table_name }}"
+      echo "Test results file path: '${{ inputs.test_results_file_path }}'"
+      echo "Test event record provided: ${{ inputs.test_event_record != '' }}"
+
+      # Show what the load input step will do
+      if [ -n "${{ inputs.test_results_file_path }}" ]; then
+        echo "Will use file path input method: ${{ inputs.test_results_file_path }}"
+      elif [ -n "${{ inputs.test_event_record }}" ]; then
+        echo "Will use inline data input method"
+      else
+        echo "INFO: No input detected - will skip CI Analytics submission (no test issues to report)"
+      fi
+      echo "-----"
+
+  - name: ensure that jq is available
+    uses: dcarbone/install-jq-action@v3.2.0
+
+  - name: load input
+    shell: bash
+    env:
+      TEMP_DIR: ${{ runner.temp }}/sts
+    run: |
+      mkdir -p $TEMP_DIR
+
+      echo "Checking inputs..."
+      echo "test_results_file_path: '${{ inputs.test_results_file_path }}'"
+      echo "test_event_record length: ${#test_event_record}"
+
+      if [ -n "${{ inputs.test_results_file_path }}" ]; then
+        # Method 1: Use provided file path
+        INPUT_FILE="${{ inputs.test_results_file_path }}"
+        echo "Using file path input method: $INPUT_FILE"
+
+        # Verify the file exists
+        if [ ! -f "$INPUT_FILE" ]; then
+          echo "ERROR: File not found: $INPUT_FILE"
+          exit 1
+        fi
+
+      elif [ -n "${{ inputs.test_event_record }}" ]; then
+        # Method 2: Create file from inline input
+        INPUT_FILE="$TEMP_DIR/input_test_event_record.jsonl"
+        echo "Using inline data input method, creating: $INPUT_FILE"
+
+        cat <<EOF > $INPUT_FILE
+      ${{ inputs.test_event_record }}
+      EOF
+
+      else
+        # No input provided - this might be okay if no flaky/failed tests exist
+        echo "INFO: No input provided."
+        echo "This could mean:"
+        echo "1. No test_results_file_path was provided, AND"
+        echo "2. No test_event_record was provided (possibly no flaky/failed tests)"
+        echo ""
+        echo "If using inline method and no flaky/failed tests exist, this is expected behavior."
+        echo "Skipping CI Analytics submission - no test issues to report."
+        echo "SKIP_SUBMISSION=true" >> $GITHUB_ENV
+        exit 0
+      fi
+
+      echo "Final INPUT_FILE: $INPUT_FILE"
+      echo "INPUT_FILE=$INPUT_FILE" >> $GITHUB_ENV
+      echo "SKIP_SUBMISSION=false" >> $GITHUB_ENV
+
+  - name: validate input for file format
+    if: env.SKIP_SUBMISSION != 'true'
```

Comment:
🔧 Consider inverting the variable name meaning and usage to avoid double-negation.

When reading this I'm thinking

> So if skipping the submission is NOT true, aka if skipping the submission is false aka if we DONT skip the submission aka if we want to submit

Maybe call it `DO_SUBMIT_METRICS` and use `if: env.DO_SUBMIT_METRICS == 'true'` etc.